packages/astro/src/assets/fonts/providers/local.ts

```diff
@@ -1,17 +1,59 @@
-import { providers } from 'unifont';
-import { defineFontProvider } from '../helpers.js';
+import type * as unifont from 'unifont';
+import type { LocalFontFamily } from '../types.js';
+import { DEFAULTS } from '../constants.js';
+import { fileURLToPath } from 'node:url';
+import { extname } from 'node:path';
+
+// TODO: move this file so it's not publicly exported
+
+// https://fonts.nuxt.com/get-started/providers#local
+// https://github.com/nuxt/fonts/blob/main/src/providers/local.ts
+// https://github.com/unjs/unifont/blob/main/src/providers/google.ts
 
 export const LOCAL_PROVIDER_NAME = 'local';
 
-export function local() {
-	return defineFontProvider({
-		name: LOCAL_PROVIDER_NAME,
-		entrypoint: 'astro/assets/fonts/providers/local',
-	});
+type InitializedProvider = NonNullable<Awaited<ReturnType<unifont.Provider>>>;
+
+type ResolveFontResult = NonNullable<Awaited<ReturnType<InitializedProvider['resolveFont']>>>;
+
+interface Options {
+	root: URL;
 }
 
-// TODO: implement
-export const provider = () =>
-	Object.assign(providers.google(), {
-		_name: 'local',
-	});
+interface ResolveOptions {
+	proxySourceURL: (value: string) => string;
+}
+
+// TODO: dev watcher and ways to update during dev
+export function createLocalProvider({ root }: Options) {
+	return {
+		resolveFont: async (
+			family: LocalFontFamily,
+			{ proxySourceURL }: ResolveOptions,
+		): Promise<ResolveFontResult> => {
+			const fonts: ResolveFontResult['fonts'] = [];
+
+			for (const src of family.src) {
+				for (const weight of src.weights ?? DEFAULTS.weights) {
+					for (const style of src.styles ?? DEFAULTS.styles) {
+						// TODO: handle fallbacks?
+						// TODO: handle subset
+						fonts.push({
+							weight,
+							style,
+							src: src.paths.map((path) => ({
+								url: proxySourceURL(fileURLToPath(new URL(path, root))),
+								// TODO: check
+								format: extname(path).slice(1),
+							})),
+						});
+					}
+				}
+			}
```

Comment:
I have mixed feelings about this matrix because it doesn't seem very efficient; it seems that for each item, we create tons of styling. Also, the RFC doesn't actually explain how the configuration is mapped to the real CSS.

I suggest you to add a paragraph that show how these configurations are eventually mapped into inline CSS, so we can have a better review of this PR